onnxruntime/core/providers/webnn/builders/helper.cc

```diff
@@ -27,6 +29,31 @@ bool GetShape(const NodeArg& node_arg, std::vector<int64_t>& shape, const loggin
   return true;
 }
 
+bool GetShapeByTensor(const onnx::TensorProto& tensor, std::vector<int64_t>& shape, const logging::Logger& logger) {
+  std::vector<uint8_t> unpacked_tensor;
+  auto status = onnxruntime::utils::UnpackInitializerData(tensor, unpacked_tensor);
+  if (!status.IsOK()) {
+    LOGS(logger, ERROR) << "Error while unpacking shape: " << status.ErrorMessage();
+    return false;
+  }
+  const auto& dims = tensor.dims();
+  if (dims.empty() || dims[0] == 0) {
+    LOGS(logger, VERBOSE) << "The shape cannot be empty.";
+    return false;
+  }
+  if (dims.size() != 1) {
+    LOGS(logger, VERBOSE) << "The shape must be 1D.";
+    return false;
+  }
+  if (tensor.data_type() != ONNXTensorElementDataType::ONNX_TENSOR_ELEMENT_DATA_TYPE_INT64) {
+    LOGS(logger, VERBOSE) << "The shape element data type must be INT64.";
+    return false;
+  }
+  const int64_t* shape_data = reinterpret_cast<const int64_t*>(unpacked_tensor.data());
+  shape = std::vector<int64_t>{shape_data, shape_data + dims[0]};
+  return true;
```

Comment:
```suggestion
  const auto& dims = tensor.dims();
  if (dims.size() != 1) {
    LOGS(logger, VERBOSE) << "The shape tensor must be 1D.";
    return false;
  }
  int64_t rank = dims[0];
  if (tensor.data_type() != ONNXTensorElementDataType::ONNX_TENSOR_ELEMENT_DATA_TYPE_INT64) {
    LOGS(logger, VERBOSE) << "The shape element data type must be INT64.";
    return false;
  }
  const int64_t* shape_data = reinterpret_cast<const int64_t*>(unpacked_tensor.data());
  shape.assign(shape_data, shape_data + rank};
  return true;
```

- If we compare dims size != 1 first, then the empty check is not needed.
- Dimensions of size 0 are legal in ONNX and represent scalar values. The WebNN EP accepts them too (I have some test cases in SimpleWebNN.html).
- Can use `assign` rather than create a new vector.

Reply:
Thanks. Please take another look at first new commit.